scala/scala.bzl

```diff
@@ -452,28 +467,36 @@ env JAVACMD=$0.runfiles/{repo}/{java} $0.runfiles/{repo}/{scala} {jvm_flags} -cl
       runfiles=runfiles)
 
 def _scala_test_impl(ctx):
-  deps = ctx.attr.deps
-  deps += [ctx.attr._scalatest_reporter]
-  jars = _collect_jars(deps)
-  (cjars, rjars) = (jars.compiletime, jars.runtime)
-  cjars += [ctx.file._scalareflect, ctx.file._scalatest, ctx.file._scalaxml]
-  rjars += [ctx.outputs.jar, ctx.file._scalalib, ctx.file._scalareflect, ctx.file._scalatest, ctx.file._scalaxml]
-  rjars += _collect_jars(ctx.attr.runtime_deps).runtime
-  _write_test_launcher(ctx, rjars)
-  return _scala_binary_common(ctx, cjars, rjars)
+    deps = ctx.attr.deps
+    deps += [ctx.attr._scalatest_reporter]
+    jars = _collect_jars(deps)
+    (cjars, rjars) = (jars.compiletime, jars.runtime)
+    cjars += [ctx.file._scalareflect, ctx.file._scalatest, ctx.file._scalaxml]
+    rjars += [
+              ctx.outputs.jar,
+              ctx.file._scalalib,
+              ctx.file._scalareflect,
+              ctx.file._scalatest,
+              ctx.file._scalaxml
+              ]
+    rjars += _collect_jars(ctx.attr.runtime_deps).runtime
+    _write_test_launcher(ctx, rjars)
+    return _scala_binary_common(ctx, cjars, rjars)
 
 _implicit_deps = {
   "_ijar": attr.label(executable=True, default=Label("@bazel_tools//tools/jdk:ijar"), single_file=True, allow_files=True),
   "_scala": attr.label(executable=True, default=Label("@scala//:bin/scala"), single_file=True, allow_files=True),
-  "_scalac": attr.label(executable=True, default=Label("@scala//:bin/scalac"), single_file=True, allow_files=True),
+  "_scalac": attr.label(cfg=HOST_CFG, executable=True, default=Label("//src/java/io/bazel/rulesscala/scalac"), allow_files=True),
```

Comment:
Well, I don't know if you noticed, but we have been compiling targets
twice. Bazel does this for the host and target compilations. I don't see
why we need to do that with JVM code like this. I wonder if this flag (in
the right spot) could change that behavior.
On Tue, Sep 13, 2016 at 06:22 ianoc <notifications@github.com> wrote:

> In scala/scala.bzl
> <https://github.com/bazelbuild/rules_scala/pull/91#discussion_r78593996>:
>
> >
> >  _implicit_deps = {
> >    "_ijar": attr.label(executable=True, default=Label("@bazel_tools//tools/jdk:ijar"), single_file=True, allow_files=True),
> >    "_scala": attr.label(executable=True, default=Label("@scala//:bin/scala"), single_file=True, allow_files=True),
> > -  "_scalac": attr.label(executable=True, default=Label("@scala//:bin/scalac"), single_file=True, allow_files=True),
> > +  "_scalac": attr.label(cfg=HOST_CFG, executable=True, default=Label("//src/java/io/bazel/rulesscala/scalac"), allow_files=True),
>
> I don't either ¯_(ツ)_/¯ ! Lifted from example when i was trying to make
> this work. Happy to drop if it all still runs. I'll check
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bazelbuild/rules_scala/pull/91/files/1b42b87a93989fa22db6657ce861a33c06b3eb03#r78593996>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEJdt-9PlCp4-1NFCORlwZMGzzqx19xks5qps23gaJpZM4J7Sy_>
> .
>